Account/FirefoxAccount.swift

```diff
@@ -190,13 +195,79 @@ open class FirefoxAccount {
             return "Not in a Token State: \(state?.label.rawValue ?? "Empty State")"
         }
     }
-
+    
+    public class FxAProfile {
+        open var avatarUrl: URL?
+        open var avatarImageData: Data?
```

Comment:
This whole thing might be more clear if you introduce an `Avatar`, which is a `URL`, a flag indicating whether the fetch succeeded, and a `UIImage`.

What happens if my avatar points to a non-image URL? Or the server is down? Or it's a 2GB TIFF?

Reply:
I'm pretty sure `UIImage` doesn't count as a view object. But it is worth checking whether there's a significant memory overhead (or win!) to using `UIImage` rather than the raw data.